src/Controls/tests/Core.UnitTests/AppThemeTests.cs

```diff
@@ -38,11 +58,31 @@ public void ThemeChangeUsingSetAppThemeColor()
 			label.SetAppThemeColor(Label.TextColorProperty, Colors.Green, Colors.Red);
 			Assert.Equal(Colors.Green, label.TextColor);
 
-			SetAppTheme(AppTheme.Dark);
+			app.LoadPage(new ContentPage { Content = label });
+
+			Assert.Equal(Colors.Green, label.TextColor);
+
+			EmulatePlatformThemeChange(AppTheme.Dark);
 
 			Assert.Equal(Colors.Red, label.TextColor);
 		}
 
+		[Fact]
+		public void ThemeChangeUsingSetAppThemeColorNonVisualElement()
+		{
+			var element = new NonVisualElement
+			{
+				Text = "Green on Light, Red on Dark"
+			};
+
+			element.SetAppThemeColor(NonVisualElement.ColorProperty, Colors.Green, Colors.Red);
+			Assert.Equal(Colors.Green, element.Color);
+
+			EmulatePlatformThemeChange(AppTheme.Dark);
+
+			Assert.Equal(Colors.Red, element.Color);
+		}
```

Comment:
Some elements, like `Shell` and `MenuItems` - which are not in the main UI hierarchy - do not need a host `Window` because they are directly connected to the `Application`. I would like to make all things not need to use the main application as this not only makes things neater but also does not require `Application.Current`. But, this isextra work for this already too-long PR.